cfe_internal/enterprise/CFE_hub_specific.cf

```diff
@@ -468,17 +474,29 @@ bundle agent cfe_internal_refresh_inventory_view
 
       "tags" slist => { "enterprise_maintenance" };
 
+  vars:
+    cfengine_enterprise_federation:am_superhub::
+      "inventory_refresh_args" -> {"ENT-4527"}
+        string => "inventory_refresh_by_hostkey $(sys.key_digest)",
+        comment => "Only refresh inventory for the superhub, other hosts get refreshed on data import";
+
+    !cfengine_enterprise_federation:am_superhub::
+      "inventory_refresh_args" string => "inventory_refresh";
+
+  methods:
+      # we need to know if we are running on a superhub
+      "superhub_config" usebundle => cfengine_enterprise_federation:config;
```

Comment:
Is the full 3 pass convergence required to know `am_superhub`?

The variables defined above, based on this bundle converging may not know about am_superhub until the next pass.

Reply:
When I run into these cases where normal order and 3 passes don't jive well, I usually break them into seperate bundles. So i might have a bundle agent inventory_refresh_args where those vars are defined, and have that called before superhub_config

Reply:
You mean **after** `superhub_config`?

Reply:
yes, sorry, something like:

```
  bundle agent cfe_internal_refresh_inventory_view
  {
    meta:
      (policy_server|am_policy_hub).enterprise_edition::
  
        "tags" slist => { "enterprise_maintenance" };

    methods:
      # Do we want to run this policy always?
      "Superhub Configuration"
        comment => "The superhub configuration policy defines classes and
                    variables needed to determine the proper way to refresh
                    inventory. We give it a full 3 pass evaluation to ensure
                    they are resolved before refreshing inventory.",
        usebundle => cfengine_enterprise_federation:config;

      "Refresh Inventory"
        comment => "Because people would like to have up to date information
                    about the state of hosts in their infrastructure.",
        usebundle => cfe_internal_refresh_inventory_view_actually;
  }
  bundle agent cfe_internal_refresh_inventory_view_actually
  {
    vars:
      cfengine_enterprise_federation:am_superhub::
        "inventory_refresh_args" -> {"ENT-4527"}
          string => "inventory_refresh_by_hostkey $(sys.key_digest)",
          comment => "Only refresh inventory for the superhub, other hosts get
                      refreshed on data import";

      !cfengine_enterprise_federation:am_superhub::
        "inventory_refresh_args" string => "inventory_refresh";

    commands:

      (policy_server|am_policy_hub).enterprise_edition::
  
        "$(sys.workdir)/httpd/php/bin/php"
          args => "$(cfe_internal_hub_vars.docroot)/index.php cli_tasks $(inventory_refresh_args)",
          contain => silent,
          comment => "This refreshes the inventory view. If the inventory view
                      is not refreshed then it will contain stale data.",
          handle  => "mpf_fresh_inventory_view",
          if => isdir( "$(cfe_internal_hub_vars.docroot)/api/modules/inventory" );
  }
```

Reply:
I chose a bit different strategy.

Reply:
I saw, pretty similar, same effect and the intention is clear.